packages/react-components/src/components/chart/utils/getInfo.ts

```diff
@@ -33,36 +46,126 @@ export const calculateTimeStamp = (xInPixel: number, widthInPixel: number, viewp
   return delta + initial;
 };
 
+export const setXWithBounds = (size: SizeConfig, x: number) => {
+  const upperLimit = size.width - DEFAULT_MARGIN;
+  const lowerLimit = DEFAULT_MARGIN;
+  return x > upperLimit ? upperLimit : x < lowerLimit ? lowerLimit : x;
+};
+
+export const getTrendCursorHeaderTimestampText = (timestampInMs: number, previousText?: string) => {
+  const title = previousText && previousText.split('\n')[0];
+  return [
+    title,
+    `{timestamp|${new Date(timestampInMs).toLocaleDateString()} ${new Date(timestampInMs).toLocaleTimeString()}}`,
+  ].join('\n');
+};
 // this returns a Graphic element of Echarts (https://echarts.apache.org/en/option.html#graphic)
 // A Trend cursor is a custom Graphic group element,
 // which has a line and other elements which gets rendered on the screen.
 // for now, we are storing the timestamp of the trend cursor in graphic element
 // which will eventually be moved to state(redux)
-export const addNewTrendCursor = (e: ElementEvent, size: SizeConfig, viewport: Viewport | undefined) => ({
-  id: `trendCursor-${uuid()}`,
-  action: '$merge',
-  type: 'group',
-  timestampInMs: calculateTimeStamp(e.offsetX, size?.width, viewport),
-  children: [
-    {
-      type: 'line',
-      z: trendCursorZIndex,
-      id: `line-${uuid()}`,
-      shape: {
-        x1: e.offsetX,
-        x2: e.offsetX,
-        y1: DEFAULT_MARGIN,
-        y2: (size?.height ?? 0) - DEFAULT_MARGIN,
+export const addNewTrendCursor = (
+  e: ElementEvent,
+  size: SizeConfig,
+  count: number,
+  graphic: InternalGraphicComponentGroupOption[],
+  setGraphic: Dispatch<SetStateAction<InternalGraphicComponentGroupOption[]>>,
+  viewport?: Viewport,
+  chart?: EChartsType
+) => {
+  const uId = uuid();
+  const timestampInMs = calculateTimeStamp(e.offsetX, size?.width, viewport);
+  const boundedX = setXWithBounds(size, e.offsetX ?? 0);
+  const newTC = {
+    id: `trendCursor-${uId}`,
+    $action: 'merge',
+    type: 'group' as const,
+    timestampInMs,
+    children: [
+      {
+        type: 'line',
+        z: trendCursorZIndex,
+        id: `line-${uId}`,
+        draggable: 'horizontal' as const,
+        shape: {
+          x1: boundedX,
+          x2: boundedX,
+          y1: DEFAULT_MARGIN,
+          y2: (size?.height ?? 0) - DEFAULT_MARGIN,
```

Comment:
Why can size be nullish?

Reply:
good call out, this was a copy paste issue, height and width is provided, so it will not be null / undefined. fixing this